### PR TITLE
[17.0][MIG] payroll: Migration to 17.0

### PR DESCRIPTION
**Pull request by hapolinario**
Migration payroll module to 17.0.
Supersedes https://github.com/OCA/payroll/pull/138

### CODE REVIEW

**Comment by nimarosa** [comment]
Will review this PR next week. Please is there are other functional or code reviewers will be great to make the process fast.

**Comment by peluko00** [comment]
@hapolinario squash commits please

**Comment by hussain** [comment]
I will do review this week.

File: payroll/views/hr_payslip_run_views.xml [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+                <sheet>
+                    <label for="name" class="oe_edit_only" />
+                    <h1>
+                        <field name="name" readonly="state != 'draft'" />
+                    </h1>
+                    <group>
+                        <group name="period_group">
+                            <label for="date_start" string="Period" />
+                            <div>
+                                <field
```
Comment (hussain) [inline payroll/views/hr_payslip_run_views.xml:139]: You are using `widget="daterange"` so there is no need to put both start & end date separately. Please remove one of them as they show duplicate on screen.

image

File: payroll/views/hr_salary_rule_category_views.xml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                            <separator string="Associated Salary Rules" />
+                            <field name="salary_rules_ids" nolabel="1" create="false">
+                                <tree>
+                                    <field name="code" decoration-bf="1" />
+                                    <field name="name" />
+                                </tree>
+                            </field>
+                        </page>
+                    </notebook>
+                    <group string="Notes">
```
Comment (hussain) [inline payroll/views/hr_salary_rule_category_views.xml:31]: The `note` field is somehow very narrow and not going full page width.

image

File: payroll/views/res_config_settings_views.xml [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        <field name="inherit_id" ref="base.res_config_settings_view_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//form" position="inside">
+                <app
+                    string="Payroll"
+                    name="payroll"
+                    groups="payroll.group_payroll_manager"
+                >
+                    <block title="Accounting" id="payroll_accountant">
+                        <setting>
```
Comment (hussain) [inline payroll/views/res_config_settings_views.xml]: There are duplicate text in configuration.

image


`<setting>` block can be replaced with:
```xml
<setting id="" string="Payroll Entries" help="Post payslips in accounting">
	<field name="module_payroll_account" />
</setting>
```


Please apply same to all other `<settings>` blocks.

File: payroll/views/hr_contribution_register_views.xml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+                    <field
+                        name="company_id"
+                        groups="base.group_multi_company"
+                        options="{'no_create': True}"
+                        class="oe_inline"
+                    />
+                </group>
+                <newline />
+                <group>
+                    <separator string="Description" />
```
Comment (hussain) [inline payroll/views/hr_contribution_register_views.xml:65]: The description box is very narrow, should be full page width probably.

image

**Comment by hussain** [comment]
Functional OK with minor view changes.

@hapolinario 
Please squash last commits of same description once you're done.

image

**Review by dreispt — APPROVED** [review]

**Review by hussain — APPROVED** [review]

**Review by celm1990 — COMMENTED** [review]
My comment a little late: I think it is not a good idea to skip `E501`. It's better to split the lines.

**Pull request merged**